File: python/roman-numerals/roman_numerals.py
VALUES = {1: 'I', 5: 'V', 10: 'X', 50: 'L', 100: 'C', 500: 'D', 1000: 'M'}


def roman(number: int) -> str:
    """Convert from decimal numbers to Roman Numerals.

    :param number: decimal number to convert
    :return: roman Numeral
    """
    output = ''
    for val in sorted(VALUES.keys(), reverse=True):
        times = number // val
        if times:
            output += VALUES[val] * times
            number = number - val*times
        sub_val = val // 5 if str(val)[0] == '5' else val // 10
        if number >= val - sub_val:
            output += VALUES[sub_val] + VALUES[val]
            number = number - (val - sub_val)

    return output

File: javascript/roman-numerals/roman-numerals.js
/** Roman Numerals exercise. */

const romanMap = new Map([
  [1000, 'M'], [900, 'CM'], [500, 'D'], [400, 'CD'], 
  [100, 'C'], [90, 'XC'], [50, 'L'], [40, 'XL'], 
  [10, 'X'], [9, 'IX'], [5, 'V'], [4, 'IV'], [1, 'I']
]);

export const toRoman = (number) => {
  let result = '';
  while (number > 0) {
    for (const [num, roman] of romanMap) {
      if (number >= num) {
        result += roman;
        number -= num;
        break;
      }
    }
  }
  return result;
};
